client.go

```diff
@@ -130,6 +134,21 @@ func (c *Client) defaultTimeout(ctx context.Context) time.Duration {
 }
 
 func (c *Client) registerProviders() {
+	// register the rpc provider
```

Comment:
since the `registerProviders()` method is getting longer, for readability, it would ideal to move these into smaller functions - `(c *Client) registerRPCProvider() error`